internal/services/cognitive/cognitive_deployment_resource.go

```diff
@@ -98,9 +100,15 @@ func (r CognitiveDeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 						ValidateFunc: validation.StringIsNotEmpty,
 					},
 
+					"source": {
```

Comment:
as this is a resource id  wouldn't it make more sense to have it be

```suggestion
					"source_resource_id": {
```